### PR TITLE
Milestone 2: Query and update animals table.

### PR DESCRIPTION
**Pull request by dfsalam**
# In this pull request I:
- ### Update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
TRANSACTION 1
- ## Inside a transaction:
- ### Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- ### Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- ### Commit the transaction.
TRANSACTION 2
- ### Verify that change was made and persists after commit.
TRANSACTION 2 AFTER COMMIT VERIFICATION
- ### Inside a transaction delete all records in the animals table, then roll back the transaction. After the rollback verify if all records in the animals table still exists. :fearful:
TRANSACTION 3 (DELETE ALL)
- ## Inside a transaction:
- ### Delete all animals born after Jan 1st, 2022.
- ### Create a savepoint for the transaction.
- TRANSACTION 4 (SAVEPOINT)
- ### Update all animals' weights to be their weight multiplied by -1.
- ### Rollback to the savepoint
- ### Update all animals' weights that are negative to be their weight multiplied by -1.
- ### Commit transaction.
TRANSACTION 4B (AFTER ROLLBACK TO SAVEPOINT)
*******EDITION**********
TRANSACTION 4C

- ## Write queries to answer the following questions:
- ### How many animals are there?
- ### How many animals have never tried to escape?
- ### What is the average weight of animals?
QUESTIONS (1-3)
- ### Who escapes the most, neutered or not neutered animals?
- ### What is the minimum and maximum weight of each type of animal?
QUESTIONS (4-5)
- ### What is the average number of escape attempts per animal type of those born between 1990 and 2000?
QUESTION 6

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -6,4 +6,16 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
 INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (3, 'Pikachu', '2021-01-07', 1, false, 15.04);
 
 INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered) VALUES (4, 'Devimon', '2017-05-12', 5, true);
+/* Milestone 2 */
+/*Add a column species of type string to your animals table. */
+ALTER TABLE animals ADD COLUMN species TEXT;
```
Comment (t-yanick) [inline data.sql]: - [ ] Great job working on this milestone. As per project requirement, everything that creates or change the tables structure should be saved in the `schema.sql` file. Kindly remove this line of code from `data.sql` to `schema.sql`
Also you can use VARCHAR for data type and set a limit: `ALTER TABLE animals ADD species varchar(255);`

File: queries.sql [hunk 2]
```diff
@@ -22,3 +22,54 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
...
+
+/*Transaction 4 */
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SELECT * FROM animals ORDER BY id;
+SAVEPOINT transaction4;
+UPDATE animals SET weight_kg = weight_kg * -1;
+SELECT * FROM animals ORDER BY id;
+ROLLBACK TO transaction4;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
```
Comment (t-yanick) [inline queries.sql:51]: - [ ] I cannot find the screenshot of the query `UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; `in your PR description. I think when you run this query correctly, all the weights should be positive values.

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @dfsalam , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Screenshots added to PR description
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (dfsalam) [reply data.sql]: Thank you. Done.

File: queries.sql (hunk 2 above)
Comment (dfsalam) [reply queries.sql:51]: Ok, I added this screenshot to the PR. Also inside the img folder as TRANSACTION 4C.jpg
TRANSACTION 4C

File: data.sql (hunk 1 above)
Comment (dfsalam) [reply data.sql]: I change the type with ALTER TABLE animals ALTER COLUMN species TYPE varchar(255);

**Review by jerryowusu — APPROVED** [review]
Hi @dfsalam , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

chelsea-mason-approve


## Highlights
Added screenshots to the PR description :heavy_check_mark: 
Great effort in making the requested changes :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**